frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -80,6 +87,38 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
                   .format(processedBranchName))
           .range(branch).create();
     }
+
+    // update the state of the .git/machete VirtualFile so that new entry is available in the VirtualFile
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> saveDocumentBeforeCheck(file));
+    // check for duplicate entries in the machete file
+    try {
+      val branchNamesFromFile = VfsUtil.loadText(file.getVirtualFile());
+      if (isBranchNameRepeated(branchNamesFromFile, processedBranchName)) {
+        holder.newAnnotation(HighlightSeverity.ERROR,
+            getNonHtmlString("string.GitMachete.MacheteAnnotator.branch-entry-already-defined")
+                .format(processedBranchName))
+            .range(branch).create();
+      }
+    } catch (PluginException | IllegalStateException ignored) { // ignore dubious IDE checks against annotation range
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @UIEffect
+  private void saveDocumentBeforeCheck(PsiFile file) {
+    val fileDocManager = FileDocumentManager.getInstance();
```

Comment:
Actually... does `fileDocManager.saveDocument(...)` actually save the file **to disk**?

Reply:
yes - `Saves the specified document to disk` is the first thing from Javadoc

Reply:
> but there is som thing wong still :c
> 
> ![image](https://user-images.githubusercontent.com/19799111/194340336-6ae38e43-722e-4fdf-b44a-4947793b3898.png)

@mkondratek - I did some more tests on my machine... And all works fine. Can you repeat your test and let me know, if the lack of highlighting occurred again?